src/platform.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 Joe Wilm, The Alacritty Project Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//! Provide operating system and platform-level integration.
+#[cfg(windows)]
+extern crate winapi;
+#[cfg(windows)]
+use std::panic;
+#[cfg(windows)]
+use std::{ffi, ptr};
+#[cfg(windows)]
+use winapi::um::winuser;
+
+#[cfg(windows)]
+fn on_windows_error(pi: &panic::PanicInfo) {
+    let msg = format!("{}\nPress Ctrl-C to Copy", pi);
+    unsafe {
+        winuser::MessageBoxA(
+            ptr::null_mut(),
+            ffi::CString::new(msg).unwrap().as_ptr(),
+            ffi::CString::new("Alacritty: Runtime Error")
+                .unwrap()
+                .as_ptr(),
+            winuser::MB_OK | winuser::MB_ICONERROR,
+        );
+    }
+}
+
+// Install a panic handler that renders the panic in a classical Windows error
+// dialog box.
+#[cfg(windows)]
+pub fn attach_windows_error_handler() {
+    panic::set_hook(Box::new(on_windows_error));

```

Comment:
`panic::set_hook` will take care of all errors. Which also means even if the user wanted to look at stderr, he couldn't do it anymore. While providing a popup is nice, it shouldn't remove the old functionality.

Using `println!` here would be a bit dangerous since there have been previous instances of that panicking when no stderr was attached. So usually something like `let _ = writeln!(io::stderr(), "{}", panic);` should do the job.

Reply:
Yes, I was thinking this exact thing, too.  It now does this.


You could say better than I, but do we care about https://github.com/rust-lang/rfcs/blob/master/text/1869-eprintln.md at all here?

Reply:
Using `eprintln` is problematic since it panics when the stderr isn't available. `let _ = writeln!(io::stderr(), "{}", panic);` does the same, but it ignores failure.

Reply:
Thank you very much for your patience.  I have learned a lot in the process (and I know how much work it is dealing with open source contributors and community having done that myself for a few projects).